src/main/java/frc/robot/subsystems/Elevator.java

```diff
@@ -64,15 +78,42 @@ public class Elevator extends SubsystemBase {
           .withKI(Constants.ElevatorConstants.I)
           .withKD(Constants.ElevatorConstants.D);
   private ElevatorSim m_elevatorSim;
+  private TunableNumber tunableP;
+  private TunableNumber tunableI;
+  private TunableNumber tunableD;
+  private Consumer<Double> consumerP;
+  private Consumer<Double> consumerI;
+  private Consumer<Double> consumerD;
+  private TunableNumber tunableMMVel;
+  private TunableNumber tunableMMAcc;
+  private TunableNumber tunableMMJerk;
+  private Consumer<Double> consumerMMVel;
+  private Consumer<Double> consumerMMAcc;
+  private Consumer<Double> consumerMMJerk;
 
   public Elevator() {
     // TODO: set up for canivore
     currentLimit.StatorCurrentLimit = 90;
     currentLimit.SupplyCurrentLimit = 70;
+    m_voltReq = new VoltageOut(0.0);
     currentLimit.SupplyCurrentLowerLimit = 40;
     currentLimit.SupplyCurrentLowerTime = 1;
     currentLimit.StatorCurrentLimitEnable = true;
     currentLimit.SupplyCurrentLimitEnable = true;
+    m_sysIdElevator =
+        new SysIdRoutine(
+            new SysIdRoutine.Config(
+                null, // Use default ramp rate (1 V/s)
+                Volts.of(2), // Reduce dynamic step voltage to 4 to prevent brownout
+                Seconds.of(3), // Use default timeout (10 s)
+                // Log state with Phoenix SignalLogger class
+                (state) -> SignalLogger.writeString("Elevator/SysIdState", state.toString())),
+            new SysIdRoutine.Mechanism(
+                (volts) -> elevatorMotor1.setControl(m_voltReq.withOutput(volts.in(Volts))),
+                null,
```

Comment:
Its initially unclear why this is null, please add a comment referencing the note about this in the WPILib docs https://docs.wpilib.org/en/stable/docs/software/advanced-controls/system-identification/creating-routine.html#routine-config